evm/src/test/java/org/hyperledger/besu/evm/code/CodeV1Test.java

```diff
@@ -367,4 +374,380 @@ private static Stream<Arguments> immediateContainsOpcodeArguments() {
             "5b".repeat(160) + "5e01ff6000")
         .map(Arguments::arguments);
   }
+
+  @ParameterizedTest(name = "{index}: {0}")
+  @MethodSource({
+    "stackEmpty",
+    "stackEmptyAtExit",
+    "stackImmediateBytes",
+    "stackUnderflow",
+    "stackRJumpForward",
+    "stackRJumpBackward",
+    "stackRJumpI",
+    "stackCallF",
+    "stackRetF",
+    "stackUnreachable"
+  })
+  void validateStackAnalysis(
+      final String ignoredName,
+      final String expectedError,
+      final int sectionToTest,
+      final List<List<Object>> rawCodeSections) {
+
+    List<CodeSection> codeSectionList = new ArrayList<>(rawCodeSections.size());
+    for (var rawCodeSection : rawCodeSections) {
+      codeSectionList.add(
+          new CodeSection(
+              Bytes.fromHexString(((String) rawCodeSection.get(0)).replace(" ", "")),
+              (Integer) rawCodeSection.get(1),
+              (Integer) rawCodeSection.get(2),
+              (Integer) rawCodeSection.get(3)));
+    }
+
+    assertThat(CodeV1.validateStack(sectionToTest, codeSectionList.toArray(new CodeSection[0])))
+        .isEqualTo(expectedError);
+  }
+
+  /**
+   * Vectors from an early <a
+   * href="https://github.com/ipsilon/eof/commit/c1a2422ddbb72db48bacd2406ed7dad28567b403#diff-ce64373581560ddf02962cb731dfa06457d5ff8615fe12b63c881ab97432f1cf">Ipsilon
+   * Prototype</a>
+   *
+   * @return parameterized test vectors
+   */
+  static Stream<Arguments> stackEmpty() {
+    return Stream.of(Arguments.of("Empty", null, 0, List.of(List.of("00", 0, 0, 0))));
```

Comment:
Just for future reference, there is an alternative way to name parameterized tests in Junit5. That would be something like the following, adding static imports for `org.junit.jupiter.params.provider.Arguments#arguments` and `org.junit.jupiter.api.Named#named`
```suggestion
    return Stream.of(arguments(named("Empty", null, 0, List.of(List.of("00", 0, 0, 0)))));
```
And you won't need the `ignoredName` field in the test.
More about this in the [Customizing Display Names](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-display-names) section of the documentation.

Reply:
named only takes one "payload" type and does not destructure  the list.  I'll revisit this when we move to Java 17.